metricbeat/module/system/filesystem/filesystem.go

```diff
@@ -0,0 +1,56 @@
+package filesystem
+
+import (
+	"github.com/elastic/beats/metricbeat/helper"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/topbeat/system"
+)
+
+func init() {
+	helper.Registry.AddMetricSeter("system", "filesystem", New)
+}
+
+// New creates new instance of MetricSeter
+func New() helper.MetricSeter {
+	return &MetricSeter{}
+}
+
+type MetricSeter struct{}
+
+func (m *MetricSeter) Setup(ms *helper.MetricSet) error {
+	return nil
+}
+
+func (m *MetricSeter) Fetch(ms *helper.MetricSet, host string) (event common.MapStr, err error) {
+
+	fss, err := system.GetFileSystemList()
+	if err != nil {
+		logp.Warn("Getting filesystem list: %v", err)
+		return nil, err
+	}
+
+	event = common.MapStr{}
+
+	for _, fs := range fss {
+		fsStat, err := system.GetFileSystemStat(fs)
+		if err != nil {
+			logp.Debug("topbeat", "Skip filesystem %d: %v", fsStat, err)
+			continue
+		}
+		system.AddFileSystemUsedPercentage(fsStat)
+
+		fsEvent := common.MapStr{
+			fsStat.DevName: system.GetFilesystemEvent(fsStat),
```

Comment:
Unfortunately organizing the data like this is making it impossible to do top like widgets in Kibana. For example top processes by memory usage, top FS by disk usage, etc. So it's a question of how uniform we want to have the data model versus enabling different viz in Kibana.

This would mean that Topbeat is not strictly a subset of Metricbeat, because the data is organized differently. This could be OK, but we have to take a conscious decision about it.

Overall I find this model in which the fields names are not predictable less flexible on the data consumption part.